django_cloud_deploy/tool_requirements.py

```diff
@@ -0,0 +1,270 @@
+
+from django_cloud_deploy.cli import io
+
+
+class UnableToAutomaticallyInstall(Exception):
+    """Thrown When the handle function of a requirement is unable to install"""
+    pass
+
+
+class MissingRequirementError(Exception):
+    """ Thrown when a requirement is missing."""
+
+    def __init__(self, name: str, how_to_install_message: str):
+        """Provides information to manually install the requirement.
+
+        Args:
+            name: str, Name of the requirement.
+            how_to_install_message: str,  Information to help user to install
+            the requirement.
+        """
+        self.name = name
+        self.how_to_install_message = how_to_install_message
+
+
+class MissingRequirementsError(Exception):
+    """ Provides the list of MissingRequirementError for check_and_handle.
+
+        This is used by the caller to provide information to the user on how to
+        manually install all the requirements missing.
+    """
+
+    def __init__(self, missing_requirements: List[MissingRequirementError]):
+        """ Contains a list of the missing requirements with information on
+            how to install all of them manually.
+
+        Args:
+            missing_requirements: List[MissingRequirementError], list of
+            missing requirements.
+        """
+        self.missing_requirements = missing_requirements
+
+
+class Requirement(object):
+    """Base class for all Requirement the tool needs to run.
+
+    First we check if a requirement has been installed
+    If it is isnt, we attempt to handle it.
+    """
+
+    @classmethod
+    def check(cls):
+        """Checks if the requirement is installed.
+
+        Raises:
+            MissingRequirementError: If the requirement is not found.
+        """
+        raise NotImplementedError
+
+    @classmethod
+    def handle(cls, console: io.IO):
+        """Attempts to install the requirement.
+
+        Raises:
+            UnableToAutomaticallyInstall: If the installation fails.
+            NotImplementedError: If we expect the user to manually install the
+            requirement.
+        """
+        raise NotImplementedError
+
+    @classmethod
+    def check_and_handle(cls, console: io.IO):
+        """ Checks if the requirement is installed.
+            If not, Attempts to install the requirement.
+            If unsuccesful, provides information on how to manully install.
+
+            Raises:
+                MissingRequirementError: If the requirement needs to be
+                installed manually.
+        """
+        try:
+            cls.check()
+        except MissingRequirementError as e:
+            try:
+                cls.handle(console)
+            except NotImplementedError:
+                raise e
+            except UnableToAutomaticallyInstall:
+                raise e
+
+
+class Gcloud(Requirement):
+    NAME = 'Gcloud SDK'
+
+    @classmethod
+    def check(cls):
+        """Checks if the requirement is installed.
```

Comment:
Change these a bit to say what the requirement is e.g.
"Checks if the Gcloud SDK is installed."""